deep_qa/layers/attention/matrix_attention.py

```diff
@@ -1,10 +1,11 @@
 from copy import deepcopy
+from typing import Dict
 
 from keras import backend as K
 from overrides import overrides
 
 from ..masked_layer import MaskedLayer
-from ...common.params import Params
+from ...common.params import pop_choice_with_default
```

Comment:
Ok I agree that I would have to look, but when I do look, the current docstrings are immediately obvious - I don't have to think at all to understand what they do, whereas in the combined method, I would have to enumerate all 4 combinations in my head before I understand how to use it. Maybe i'm being picky but I think it's easier.